py/runtime.c

```diff
@@ -1416,10 +1416,9 @@ void *m_malloc_fail(size_t num_bytes) {
     } else if (gc_is_locked()) {
         mp_raise_msg(&mp_type_MemoryError, "memory allocation failed, heap is locked");
     #endif
-    } else {
-        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_MemoryError,
-            "memory allocation failed, allocating %u bytes", (uint)num_bytes));
     }
+    nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_MemoryError,
+        "memory allocation failed, allocating %u bytes", (uint)num_bytes));
```

Comment:
While you are at it you may as well remove the `if (0)` and change the `} else if (...) { ` to `if (...) {`